File: typescript/packages/bench/src/rules/grossProfitRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class GrossProfitRule implements RuleContract {
    readonly key = "gross-profit";

    evaluate(data: Dataset): boolean {
        return data.revenueCents() > data.partsCostCents();
    }
}

File: rust/crates/atelier-bench/src/rules/gross_profit_rule.rs
//! Rule 21/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct GrossProfitRule;

impl RuleContract for GrossProfitRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        data.revenue_cents() > 0
    }
}
